### ISSUE TITLE
Implicit override "PropertyItemBase" Target of PropertyGrid

### ISSUE THREAD

**Issue by xceedsoftware** [xceedsoftware/wpftoolkit#635]
<b>dsscsystems[CodePlex]</b> <br />I am having problem, i am sort of new to WPF and have looked up many examples, i even added the Style above and below the declaration of the PropertyGrid Style. I have added the resource file in attachments what am i doing wrong?
ltStyle TargetType=quot{x:Type pg:PropertyGrid}quotgt
        ltStyle.Resourcesgt
           lt!- tried added here still can't overrride the default internal quotPropertyItemBasequot Style for some reason.
              ltStyle TargetType=quot{x:Type pg:PropertyItemBase}quot BasedOn=quot{StaticResource {x:Type pg:PropertyItemBase}}quotgt
               lt/Stylegt
        lt/Style.Resourcesgt

................................................
lt/Stylegt

**Comment by xceedsoftware** [comment]
<b>emartin[CodePlex]</b> <br />The reason why your style was not working, is because implicit styles only work when they target instance specific classes. Since PropertyItemBase is the base class for PropertyItem, It won't be applied.


In order to work you need tho set the TargetType to PropertyItem:
Style TargetType={x:Type pg:PropertyGrid}
        Style.Resources
           !- Here target PropertyItem --
              Style TargetType={x:Type pg:PropertyItem} BasedOn={StaticResource {x:Type pg:PropertyItemBase}}
               /Style
        /Style.Resources

................................................
/Style

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Yes, I declared it in the 
Style.Resources 
from 
Style TargetType={x:Type pg:PropertyGrid}
and it worked.

**Comment by xceedsoftware** [comment]
<b>dsscsystems[CodePlex]</b> <br />So it worked for u? U declared it inside of the PropertyGrid.Resource/Property.Resource Tab and it worked for u?

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />From what I understand, you are trying to use a new style for the PropertyItem object.

To test, I removed the 
Style TargetType={x:Type pg:PropertyItemBase}
style from the PropertyGrid/Themes/Aero2.NormalColor.xaml (Generic.xaml if you are using Windows7),
and replaced it with the 
Style TargetType={x:Type pg:PropertyItemBase}
from your file in attachement. I used a Green background to make sure it is different.


Then, using the code you submitted :
Style TargetType={x:Type pg:PropertyGrid}
    Style.Resources
          Style TargetType={x:Type pg:PropertyItemBase} BasedOn={StaticResource {x:Type pg:PropertyItemBase}}
           /Style
    /Style.Resources

................................................
/Style 

In the 
style TargetType={x:Type pg:PropertyGrid}. 

Result is a new display for the PropertyItem. What exactly is not working ?
